src/dscom/names/ComAliasNameResolver.cs

```diff
@@ -24,7 +24,7 @@ internal sealed class ComAliasNameResolver : INameResolver
 
     public ComAliasNameResolver(Assembly assembly)
     {
-        var comVisibleTypes = assembly.GetTypes().Where(t => t.IsPublic && t.GetCustomAttribute<ComVisibleAttribute>() != null);
+        var comVisibleTypes = assembly.GetLoadableTypes().Where(t => t.IsPublic && t.GetCustomAttribute<ComVisibleAttribute>() != null);
```

Comment:
I'll take a closer look at this problem tomorrow. At the moment I don't see an obvious way to solve it.

Reply:
My first idea is that the `ComAliasNameResolver` needs access to the `TypeLibExporterNotifySink` instance to report a warning event with `ReportEvent`.  
The name resolver doesn't really need the WriterContext if it can use the event structure. 
Maybe a change to the `INameResolver` interface to include a property for the event sink.     
And giving `AssemblyExtensions` a 
`internal static IEnumerable<Type> GetLoadableTypesAndLog(this Assembly assembly, ITypeLibExporterNotifySink? sink)`  method where warning events are fired. 

Sorry for the long-winded rambling. I haven't tried it out yet.

Reply:
I agree with your assessment.  The comment was misworded.  The primary intent of the PR is to suppress the crash.  There is an existing warning generated by `WriterContext`, but there is no guarantee that the ignored types in `ComAliasNameResolver` are the same as those which have a warning generated from `WriterContext`.

Reply:
Hi @mgaffigan  

I took some time to try things out, but I noticed that we did some strange stuff with the order in which the `NameResolver` and `TypeLibExporterNotifySink` objects are created and the notification sink does not exist yet at the point the assembly is scanned first and you are calling the `GetLoadableTypes` method.  

That's a limitation #206 we will have to fix at some other time. 

Given that the warning about being unable to load the `PresentationFramework` is also output later that's something we can live with it.